### PR TITLE
Convert validate to a visitor

### PR DESCRIPTION
**Pull request by rjoleary**
Signed-off-by: Ryan O'Leary <ryanoleary@google.com>

### CODE REVIEW

File: pkg/uefi/file.go [hunk 1]
```diff
@@ -178,23 +178,24 @@ func (a *fileAttr) setLarge(large bool) {
 }
 
 // Checks if we need to checksum the file body
```
Comment (rminnich) [inline pkg/uefi/file.go]: HasChecksum checks ...

File: pkg/uefi/file.go [hunk 2]
```diff
@@ -157,7 +157,7 @@ type FileHeader struct {
 }
 
 // Checks if the large file attribute is set
```
Comment (rminnich) [inline pkg/uefi/file.go]: IsLarge checks ...

File: pkg/uefi/file.go [hunk 3]
```diff
@@ -178,23 +178,24 @@ func (a *fileAttr) setLarge(large bool) {
...
 
 // Checks if we need to checksum the file body
-func (a fileAttr) hasChecksum() bool {
+func (a fileAttr) HasChecksum() bool {
 	return a&0x40 != 0
 }
 
 // HeaderLen is a helper function to return the length of the file header
 // depending on the file size
 func (f *File) HeaderLen() uint64 {
```
Comment (rminnich) [inline pkg/uefi/file.go]: I think you can drop 'is a helper function'
HeaderLen returns the length of the file header.

File: pkg/visitors/validate.go [hunk 4]
```diff
@@ -0,0 +1,270 @@
...
+	"fmt"
+	"io"
+	"os"
+
+	"github.com/linuxboot/fiano/pkg/uefi"
+)
+
+// Validate performs extra checks on the firmware image.
+type Validate struct {
+	// In
```
Comment (rminnich) [inline pkg/visitors/validate.go]: hmm. Not sure I like these In and Out comments ....

File: pkg/uefi/file.go (hunk 1 above)
Comment (rjoleary) [reply pkg/uefi/file.go]: I'll fix it, but it's true the linter has no problem, even with:

golint -min_confidence=0.0 file.go

**Pull request merged**